docs/explanations/architecture/styles.md

```diff
@@ -0,0 +1,143 @@
+
+On the other hand, there's a number of stylesheets being loaded in the front end:
+
+- **Blocks**. The stylesheets that come with the block. In the front end, you can find a single stylesheet with all block styles defined by WordPress (`wp-block-library-*` ) or separate stylesheets per block in use (as in `wp-block-group-*`, `wp-block-columns-*`, etc). See [this note](https://make.wordpress.org/core/2021/07/01/block-styles-loading-enhancements-in-wordpress-5-8/) for the full details.
+- **Global styles**. These styles are generated on the fly by using data coming from a theme.json file: see [note](https://make.wordpress.org/core/2021/06/25/introducing-theme-json-in-wordpress-5-8/), [reference](https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/), and [how to guide](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/). Specifically, it merges the contents of the theme.json coming with WordPress, the theme.json coming with the theme if any, and the user data provided via the global styles sidebar in the site editor. The result of processing this data is visible in the  `global-styles-inline-css` stylesheet.
+- **Theme**. Historically, the themes enqueued their own stylesheets, whose id would be the theme name, as in `twentytwentytwo-style-css`. In addition to having its own styesheets they can now declare a theme.json file whose styles will be part of the global stylesheet.
+- **Other**. WordPress and plugins can also enqueue stylesheets.
+
+### 2. User-provided block styles
+
+Since the introduction of the block editor in WordPress 5.0, there were tools for the users to "add styles" to specific blocks. By using these tools, the user would attach new classes or inline styles to the blocks, modifying their visual aspect.
+
+By default, blocks come with a given HTML markup. Think of the paragraph block, for example:
+
+```html
+<p></p>
+```
+
+In its simplest form, any style rule that targets the `p` selector will apply styles to this block, whether it comes from a block, a theme, etc.
+
+The user may change the state of this block by applying different styles: a text alignment, a color, a font size, a line height, etc. These states are reflected in the HTML markup of the block in the form of HTML attributes, mainly through the `class` or `style` attributes, though it can be any other the block author sees fit.
+
+After some user modifications to the block, the initial markup may become something like this:
+
+```html
+<p
+	class="has-color has-green-color has-font-size has-small-font-size my-custom-class"
+	style="line-height: 1em"
+></p>
+```
+
+This is what we refer to as "user-provided block styles". Other names people use are local styles or serialized styles. Essentially, each tool (font size, color, etc) ends up adding some classes and/or inline styles to the block markup. The content of these classes is part of the block, global, or theme stylesheets.
```

Comment:
Totally optional, but "content" made me think of markup rather than CSS, so I was wondering if the following would work:

```suggestion
This is what we refer to as "user-provided block styles". Other names people use are local styles or serialized styles. Essentially, each tool (font size, color, etc) ends up adding some classes and/or inline styles to the block markup. The CSS styling for these classes is part of the block, global, or theme stylesheets.
```